tests/application_tests/src/test_report.py

```diff
@@ -89,32 +98,31 @@ def test_login_and_logout(self):
     def test_add_report(self):
         """Test that a logged-in user can add a report."""
         self.login()
-        nr_reports = len(self.driver.find_elements(By.CLASS_NAME, "card"))
+        nr_cards = len(self.dashboard_cards())
```

Comment:
Hmm, but these renames would suggest the test be called `test_add_card`. It's kinda sneaky because we never really refer to cards as cards in the UI, they were even read-only reports :D The "button.primary" is even sneaky because there are visually two buttons, but the second one isn't actually a html button element! Oh well, I guess functional/application tests are usually a bit dirty - making the variable naming more consistent here probably isn't doable without a refactor.

Reply:
Not all cards are reports, e.g. the Legend card. I'll add a comment to explain.